commands_buy.go

```diff
@@ -58,5 +59,49 @@ func (b buyCmd) ToAuditEntry() string {
 }
 
 func (b buyCmd) Execute() {
-	consoleLog.Warning("Not implemented: BUY")
+	consoleLog.Debug("Starting Buy")
+	if _, accountExists := accountStore[b.userID]; !accountExists {
+		consoleLog.Errorf("No account for %s exists", b.userID)
+		return
+	}
+
+	userAccount := accountStore[b.userID]
+
+	if userAccount.balance.ToFloat() < b.amount.ToFloat() {
```

Comment:
> The user's account must be greater or equal to the amount of the purchase. 

What "the amount of the purchase" is is ambiguous (i.e. amount passed in command or how much is needed to fulfill order). We should clarify in class tomorrow.